### ISSUE TITLE
Possible memory leak

### ISSUE THREAD

**Issue by XakepSDK** [ValveSoftware/steam-for-linux#6274]
#### Your system information

* Steam client version (build number or date): May 15 2019, at 21:17:00
* Distribution (e.g. Ubuntu): Arch Linux
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes

Root process `steam` uses more than 500MiB RAM. This number doesn't go down.
After restart steam uses ~70MiB for root process, ~41MiB for child `steam` process and ~100MiB for all `steamwebhelper`.

I think this is somehow related to downloading system. Numbers going up and down after every download, but looks it doesn't clear everything properly.

**Comment by ptkato** [comment]
It happens here to, sometimes steam uses over 750MiB and becomes very irresponsible while doing so, restarting the client fixes the problem.

**Comment by evilynux** [comment]
Something is CLEARLY pointing that direction, at least with the current beta version of the client. 
From this morning:
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND
14085 evilynux  20   0 2588552   1,6g  55952 R   3,0  20,3   2:28.16 steam                                                                                   
14249 evilynux  20   0 1479560 824548  89356 S  41,5  10,2   2:12.46 Borderlands2

So the steam client consumes little under 1.6 GB of RAM (RES-SHR) whereas Borderlands2 consumes about half of that. Looks weird to me.

**Comment by Lukypie** [comment]
For me it's even worse, it's taking 1.5GB right now.
![image](https://user-images.githubusercontent.com/7904559/60801590-8bca9380-a177-11e9-938c-e0ca6ce03fcd.png)

**Comment by StephenLynx** [comment]
Can confirm, using ubuntu 18. Just caught it using 1.4gb.

**Comment by hmschreck** [comment]
Can also confirm this issue - within a few hours, it's up to 1GB of RAM utilized.

**Comment by jnines** [comment]
Can also confirm.  Using Arch with beta client (although the standard client had the same issue).  Sometimes it's only using ~65MB, sometimes well over 500MB.

**Comment by Amathadius** [comment]
I just force kill steamwebhelper process because it eats **10BG** of ram.

Archlinux
(New) Beta Steam Client

**Comment by XakepSDK** [comment]
@Amathadius looks like it's another bug, disable GPU accel in settings so it won't leak.

**Comment by kisak-valve** [comment]
Hello @Amathadius, sounds like you've encountered the memory leak being tracked at #6502, not the leak being tracked here.

**Comment by Amathadius** [comment]
I see, thanks

**Comment by jnines** [comment]
Yes it looks like they fixed the other bug, whatever this is is different and still exists.
![steam](https://user-images.githubusercontent.com/51514842/66001989-4bbe3880-e468-11e9-8ddd-d6613bb3e1b1.png)

**Comment by telans** [comment]
I'm having the same issue with `steamwebhelper`. I don't assume valgrind will help show anything useful (proprietary, no debug symbols I assume)?

![Screenshot_20211130_234449](https://user-images.githubusercontent.com/45819748/144033141-1c3204ee-1018-40b8-921f-0c3f2e1e9846.png)
